File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */
// todo
// @lc code=start
func twoSum(nums []int, target int) []int {
	index := make(map[int]int, len(nums))
	for i, b := range nums {
		if j, ok := index[b]; ok {
			return []int{j, i}
		}
		index[target-b] = i
	}
	return nil
}
// @lc code=end

File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include <vector>
#include <unordered_map>
using namespace std;
// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> cache;
        if (nums.size()<2)
        {
            return {};
        }
        for (int i = 0; i < nums.size(); i++)
        {
            if (cache.count(nums[i])!=0)
            {
                return {cache[nums[i]], i};
            }
            cache.emplace(target-nums[i], i);
        }
        return {};
    }
};
// @lc code=end

